flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectStreamSink.java

```diff
@@ -30,17 +32,19 @@
 @Internal
 public class CollectStreamSink<T> extends DataStreamSink<T> {
 
-    private final LegacySinkTransformation<T> transformation;
+    private final PhysicalTransformation<T> transformation;
 
+    @SuppressWarnings("unchecked")
     public CollectStreamSink(DataStream<T> inputStream, CollectSinkOperatorFactory<T> factory) {
         super(inputStream, (CollectSinkOperator<T>) factory.getOperator());
         this.transformation =
-                new LegacySinkTransformation<>(
-                        inputStream.getTransformation(), "Collect Stream Sink", factory, 1);
+                (PhysicalTransformation<T>)
```

Comment:
Why is the cast necessary?
Can you try to make `public class LegacySinkTransformation<T> extends PhysicalTransformation<T>`?
The second ctor should a factory method instead...